docs/general/architecture.md

```diff
@@ -0,0 +1,128 @@
+    SLJIT_R0, 0, SLJIT_R0, 0);
+// The result of the instruction is undefined.
+// Even crash is possible for some instructions
+// (e.g. on MIPS-64).
+```
+
+However, it is always allowed to overwrite a register regardless of its previous value:
+
+```c
+sljit_emit_op1(compiler, SLJIT_MOV,
+    SLJIT_R0, 0, SLJIT_MEM1(SLJIT_R1), 0);
+// An intptr_t value is loaded into SLJIT_R0
+sljit_emit_op1(compiler, SLJIT_MOV32,
+    SLJIT_R0, 0, SLJIT_MEM1(SLJIT_R2), 0);
+// From now on SLJIT_R0 contains an int32_t
+// value. The previous value is discarded.
+```
+
+Type conversion instructions are provided to convert an `int32_t` value to an `intptr_t` value and vice versa.
+In certain architectures these conversions are *nops* (no instructions are emitted).
+
+#### Memory Accessing
+
+Register arguments of `SLJIT_MEM1` / `SLJIT_MEM2` addressing modes must contain `intptr_t` data.
+
+#### Signed / Unsigned Values
+
+Most operations are executed in the same way regardless of if the value is signed or unsigned.
+These operations have only one instruction form (e.g. `SLJIT_ADD` / `SLJIT_MUL`).
+Instructions where the result depends on the sign have two forms (e.g. integer division, long multiply).
+
+### Floating Point Registers
+
+Floating point registers can either contain a single or double precision value.
+Similar to integer registers, the data type of the value stored in a source register must match the data type expected by the instruction.
+Otherwise the result is undefined (even a crash is possible).
+
+#### Rounding
+
+Similar to standard C, floating point computation results are rounded toward zero.
+
+### Boolean Status Flags
+
+Conditional branches usually depend on the value of CPU status flags.
+These status flags are boolean values and can be set by certain instructions.
+
+To achive maximum efficiency and portability, the following rules were introduced:
+  - Most instructions can freely modify these status flags except if `SLJIT_KEEP_FLAGS` is passed.
+  - The `SLJIT_KEEP_FLAGS` option may have a performance overhead, so it should only be used when necessary.
+  - The `SLJIT_SET_E`, `SLJIT_SET_U`, etc. options can force an instruction to correctly set the specified status flags. However, all other status flags are undefined. This rule must always be kept in mind!
+  - Status flags cannot be controlled directly (there are no set / clear / invert operations).
+
+The last two rules allow efficient mapping of status flags.
+For example the arithmetic and multiply overflow flags are mapped to the same overflow flag bit on `x86`.
+This is allowed, since no instruction can set both of these flags.
+When either of them is set by an instruction, the other can have any value (this satisfies the *"all other flags are undefined"* rule).
+Therefore mapping two SLJIT flags to the same CPU flag is possible.
+Even though SLJIT supports a dozen status flags, they can be efficiently mapped to CPUs with only 4 status flags (e.g. `ARM` or `SPARC`).
```

Comment:
This is outdated. Now sljit supports only two status flags: zero and custom. Zero represents that the result is zero, custom can be set by various flag setter option.